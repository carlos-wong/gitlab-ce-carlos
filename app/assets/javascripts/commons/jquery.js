import 'jquery';

// common jQuery plugins
import 'jquery-ujs';
import 'vendor/jquery.endless-scroll';
import 'jquery.caret'; // must be imported before at.js
import 'at.js';
import 'vendor/jquery.scrollTo';
import 'jquery.waitforimages';
