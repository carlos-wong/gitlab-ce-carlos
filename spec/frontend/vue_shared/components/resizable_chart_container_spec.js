import Vue from 'vue';
import { mount } from '@vue/test-utils';
import ResizableChartContainer from '~/vue_shared/components/resizable_chart/resizable_chart_container.vue';
import $ from 'jquery';

jest.mock('~/lib/utils/common_utils', () => ({
  debounceByAnimationFrame(callback) {
    return jest.spyOn({ callback }, 'callback');
  },
}));

describe('Resizable Chart Container', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(ResizableChartContainer, {
      attachToDocument: true,
      scopedSlots: {
        default: `
        <div class="slot" slot-scope="{ width, height }">
          <span class="width">{{width}}</span>
          <span class="height">{{height}}</span>
        </div>
        `,
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('updates the slot width and height props', () => {
    const width = 1920;
    const height = 1080;

    // JSDOM mocks and sets clientWidth/clientHeight to 0 so we set manually
    wrapper.vm.$refs.chartWrapper = { clientWidth: width, clientHeight: height };

    $(document).trigger('content.resize');

    return Vue.nextTick().then(() => {
      const widthNode = wrapper.find('.slot > .width');
      const heightNode = wrapper.find('.slot > .height');

      expect(parseInt(widthNode.text(), 10)).toEqual(width);
      expect(parseInt(heightNode.text(), 10)).toEqual(height);
    });
  });

  it('calls onResize on manual resize', () => {
    $(document).trigger('content.resize');
    expect(wrapper.vm.debouncedResize).toHaveBeenCalled();
  });

  it('calls onResize on page resize', () => {
    window.dispatchEvent(new Event('resize'));
    expect(wrapper.vm.debouncedResize).toHaveBeenCalled();
  });
});
