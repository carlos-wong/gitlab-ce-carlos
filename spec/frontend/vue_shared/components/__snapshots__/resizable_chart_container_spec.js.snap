// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resizable Chart Container renders the component 1`] = `
<div>
  <div
    class="slot"
  >
    <span
      class="width"
    >
      0
    </span>
     
    <span
      class="height"
    >
      0
    </span>
  </div>
</div>
`;
